ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1020,7 +1020,7 @@ object JdbcLedgerDao {
         |on conflict (deduplication_key)
         |  do update
         |  set deduplicate_until={deduplicateUntil}
-        |  where pcs.deduplicate_until < {submittedAt}""".stripMargin
+        |  where pcs.deduplication_key = {deduplicationKey} and pcs.deduplicate_until < {submittedAt}""".stripMargin
```

Comment:
Is this part necessary? AFAIK `ON CONFLICT DO UPDATE` only updates the conflicting row, and since the conflict was on `deduplication_key` the newly inserted condition should always be true.

Reply:
Yes, you are right, @rautenrieth-da. This is unnecessary and I will remove it.